src/inc/msquic.h

```diff
@@ -269,6 +275,8 @@ typedef struct QUIC_CREDENTIAL_CONFIG {
     const char* Principal;
     void* Reserved; // Currently unused
     QUIC_CREDENTIAL_LOAD_COMPLETE_HANDLER AsyncHandler; // Optional
+    QUIC_CIPHER_SUITE* AllowedCipherSuites;             // Optional
+    uint8_t AllowedCipherSuitesLength;
```

Comment:
`QUIC_CREDENTIAL_FLAGS Flags;` needs to control read access to this field so as to not break backwards compatibility.

Reply:
I am also leaning towards prefering this be a set of flags instead of an array of enums. It's more compact, we don't need that many config options, and it's simpler for an app to just OR some flags than build up an array (of which you have to worry about memory lifetime).

Reply:
The one thing we lose with moving to flags from an array of enums is the caller can no longer specify a preferred order of cipher suites; we determine the order when we build the string for OpenSSL.
Schannel doesn't allow for caller to specify order either; only whether it is allowed or not.
The .NET documentation for [CipherSuitesPolicy](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.ciphersuitespolicy.-ctor?view=netcore-3.1#System_Net_Security_CipherSuitesPolicy__ctor_System_Collections_Generic_IEnumerable_System_Net_Security_TlsCipherSuite__)  doesn't guarantee the client-specified order is respected either, so this might be a moot point.

Reply:
Yeah... Let's just order by most secure.